packages/tests/src/tests/test_p2ms.cairo

```diff
@@ -0,0 +1,109 @@
+    let raw_transaction = hex_to_bytecode(@raw_transaction_hex);
+    let transaction = EngineInternalTransactionTrait::deserialize(raw_transaction);
+    let utxo_hints = array![prev_out];
+
+    let res = validate::validate_p2ms(@transaction, 0, utxo_hints);
+    assert!(res.is_ok(), "P2MS 3-of-3 transaction validation failed");
+}
+
+#[test]
+fn test_p2ms_1of1() {
+    // Test case for 1-of-1 multisig
+    let prevout_pk_script_1of1 =
+        "0x514104cc71eb30d653c0c3163990c47b976f3fb3f37cccdcbedb169a1dfef58bbfbfaff7d8a473e7e2e6d317b87bafe8bde97e3cf8f065dec022b51d11fcdd0d348ac4410461cbdcc5409fb4b4d42b51d33381354d80e550078cb532a34bfa2fcfdeb7d76519aecc62770f5b0e4ef8551946d8a540911abe3e7854a26f39f58b25c15342af52ae";
+    let prev_out_1of1 = UTXO {
+        amount: 1000000,
+        pubkey_script: hex_to_bytecode(@prevout_pk_script_1of1),
+        block_height: 164467
+    };
+
+    let raw_transaction_hex =
+        "0x0100000001b14bdcbc3e01bdaad36cc08e81e69c82e1060bc14e518db2b49aa43ad90ba26000000000490047304402203f16c6f40162ab686621ef3000b04e75418a0c0cb2d8aebeac894ae360ac1e780220ddc15ecdfc3507ac48e1681a33eb60996631bf6bf5bc0a0682c4db743ce7ca2b01ffffffff0140420f00000000001976a914660d4ef3a743e3e696ad990364e555c271ad504b88ac00000000";
+    let raw_transaction = hex_to_bytecode(@raw_transaction_hex);
+    let transaction = EngineInternalTransactionTrait::deserialize(raw_transaction);
+    let utxo_hints = array![prev_out_1of1];
+
+    let res = validate::validate_p2ms(@transaction, 0, utxo_hints);
+    assert!(res.is_ok(), "P2MS 1-of-1 multisig validation failed");
+}
+
+#[test]
+fn test_p2ms_1of2_invalid_pubkey() {
+    // Test case for 1-of-2 multisig with an invalid public key
+    let prevout_pk_script_1of2_invalid =
+        "0x5121037953dbf08030f67352134992643d033417eaa6fcfb770c038f364ff40d7615882100bd2fda4cf456d64386a0756f580101a607c25bd8d6814693bdf16e2a7ba3e45c52ae";
+    let prev_out_1of2_invalid = UTXO {
+        amount: 92750000,
+        pubkey_script: hex_to_bytecode(@prevout_pk_script_1of2_invalid),
+        block_height: 229517
+    };
+
+    let raw_transaction_hex =
+        "0x0100000001b14bdcbc3e01bdaad36cc08e81e69c82e1060bc14e518db2b49aa43ad90ba26000000000490047304402203f16c6f40162ab686621ef3000b04e75418a0c0cb2d8aebeac894ae360ac1e780220ddc15ecdfc3507ac48e1681a33eb60996631bf6bf5bc0a0682c4db743ce7ca2b01ffffffff0140420f00000000001976a914660d4ef3a743e3e696ad990364e555c271ad504b88ac00000000";
+    let raw_transaction = hex_to_bytecode(@raw_transaction_hex);
+    let transaction = EngineInternalTransactionTrait::deserialize(raw_transaction);
+    let utxo_hints = array![prev_out_1of2_invalid];
+
+    let res = validate::validate_p2ms(@transaction, 0, utxo_hints);
+    assert!(res.is_err(), "P2MS 1-of-2 with invalid public key should fail validation");
+}
```

Comment:
There is also a 2-of-3 ms example and a 20-of-20 example that would be good to have tests for if you can.

Reply:
Okay boss will add this